cpp/src/io/utilities/column_buffer.cpp

```diff
@@ -181,127 +183,161 @@ std::unique_ptr<column> make_column(column_buffer_base<string_policy>& buffer,
                                  stream,
                                  buffer._mr);
-    } break;
+      } break;
+
+      case type_id::STRUCT: {
+        std::vector<std::unique_ptr<cudf::column>> output_children;
+        output_children.reserve(buffer.children.size());
+        for (size_t i = 0; i < buffer.children.size(); ++i) {
+          column_name_info* child_info = nullptr;
+          if (schema_info != nullptr) {
+            schema_info->children.emplace_back("");
+            child_info = &schema_info->children.back();
+          }
+
+          CUDF_EXPECTS(not schema.has_value() or schema->get_num_children() > i,
+                       "Invalid schema provided for read, expected more child data for struct!");
+          auto const child_schema = schema.has_value()
+                                      ? std::make_optional<reader_column_schema>(schema->child(i))
+                                      : std::nullopt;
+
+          output_children.emplace_back(
+            construct_column(buffer.children[i], child_info, child_schema));
+        }
 
-    default: {
-      return std::make_unique<column>(buffer.type,
-                                      buffer.size,
-                                      std::move(buffer._data),
-                                      std::move(buffer._null_mask),
-                                      buffer._null_count);
+        return make_structs_column_unsanitized(buffer.size,
+                                               std::move(output_children),
+                                               buffer._null_count,
+                                               std::move(buffer._null_mask),
+                                               stream,
+                                               buffer._mr);
+      } break;
+
+      default: {
+        return std::make_unique<column>(buffer.type,
+                                        buffer.size,
+                                        std::move(buffer._data),
+                                        std::move(buffer._null_mask),
+                                        buffer._null_count);
+      }
+    }
+  };
+
+  auto col = construct_column(buffer, schema_info, schema);
+  if (buffer.type.id() == type_id::STRUCT) {
+    if (col->nullable()) {
+      auto col_contents = col->release();
+      for (auto& child : col_contents.children) {
+        child = structs::detail::superimpose_nulls(
+          static_cast<bitmask_type const*>(col_contents.null_mask->data()),
+          buffer._null_count,
+          std::move(child),
+          stream,
+          buffer._mr);
+      }
+
+      return std::make_unique<column>(
+        cudf::data_type{type_id::STRUCT},
+        buffer.size,
+        rmm::device_buffer{},  // Empty data buffer. Structs hold no data.
+        std::move(*col_contents.null_mask),
+        buffer._null_count,
+        std::move(col_contents.children));
     }
   }
+
+  return std::move(col);
```

Comment:
Don't need `std::move()` when return. In addition, it should better be named `output`.
```suggestion
  return output;
```